task/dump-agent/src/udp.rs

```diff
@@ -0,0 +1,167 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{ServerImpl, DUMP_READ_SIZE};
+use hubpack::SerializedSize;
+use ringbuf::*;
+use task_net_api::*;
+
+static_assertions::const_assert_eq!(
+    DUMP_READ_SIZE,
+    humpty::udp::DUMP_READ_SIZE
+);
+
+#[derive(Copy, Clone, PartialEq)]
+enum Trace {
+    None,
+    DeserializeError(hubpack::Error),
+    DeserializeHeaderError(hubpack::Error),
+    SendError(SendError),
+    WrongVersion(u8),
+}
+
+ringbuf!(Trace, 16, Trace::None);
+
+// We are sending a (Header, Result<Response, Error>) to the host
+const MAX_UDP_TX_SIZE: usize = <(
+    humpty::udp::Header,
+    Result<humpty::udp::Response, humpty::udp::Error>,
+)>::MAX_SIZE;
```

Comment:
It might be cleaner to define structs in `humpty` that contain both the header and request/response instead of using tuples. (I assume as written humility will have to repeat this constant and also serialize/deserialize into tuples?)